.gitignore

```diff
@@ -1,32 +1,28 @@
-# Prerequisites
-*.d
```

Comment:
Why did the ignored C++ parts get removed?

Reply:
I removed them because all of those should be in the build directory which is ignored. We can add them back, but I figured I'd try to slim it down as much as possible

Reply:
Not totally sure how the Python build process will work yet, but will that be true for the dynamic libraries too? I had assumed that file would be copied into the python directory upon building, but I'm not sure on the pybind11 build details yet. I guess we can leave it out and add it back in later if a need arises